docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md

```diff
@@ -0,0 +1,663 @@
+
+1. Ensure you have backup first, then remove the data node state: `rm -r <VEGA-NETWORK-HOME>/state/data-node/`
+2. Save a backup of your old data node config: `cp <VEGA-NETWORK-HOME>config/data-node/config.toml <BACKUP-FOLDER>/data-node-config.toml`
+3. Initiate the data node config: `<VEGA-BIN> datanode init --archive --home=<VEGA-NETWORK-HOME> <CHAIN-ID> --force`
+4. Modify generated data node config: `<VEGA-NETWORK-HOME>config/data-node/config.toml` - see section below for important parameters
+
+Use the `--archive` flag to keep all data, forever. Read more about the [data retention modes](../../concepts/vega-chain/data-nodes.md#data-retention-modes) available.
+
+#### Data node config
+Important config keys that you need to update: 
+
+- `AutoInitialiseFromNetworkHistory` - We recommend setting it to `false` when you start the network from scratch (e.g. checkpoint restart), or if there is no other data node available.
+- `ChainID` - Make sure it matches the new chain ID for your network.
+- `Admin.Server.SocketPath` - Path for the Unix `sock` file; Ensure parent directory exists. Example may be `<VEGA-NETWORK-HOME>/run/datanode.sock`.
+- `API.CoreNodeIP` - IP of the server where the Vega core node is running. Often, it is `127.0.0.1.` (localhost).
+- `API.CoreNodeGRPCPort` - Port you expose for the Vega core gRPC node. Default: `3002`.
+- `SQLStore.wipe_on_startup` - Defines if the data node removes data from the postgreSQL after the restart. We recommend setting it to `false`.
+- `SQLStore.UseEmbedded` - If true, internal (managed by the binary itself) postgreSQL is used. We strongly recommend setting it to `false` for production, as it is intended for testing only.
+- `SQLStore.ConnectionConfig` - Update the entire section, as it is where you set PostgreSQL credentials.
+- `NetworkHistory.Enabled` - Enables IPFS network history. See the [network history](#a-bit-about-network-history) section below.
+
+#### Example of the PostgreSQL connection section:
+
+```toml
+  [SQLStore.ConnectionConfig]
+    Host = "localhost"
+    Port = 5432
+    Username = "<VEGA-DB-USER>"
+    Password = "<VEGA-DB-PASS>"
+    Database = "<VEGA-DB-NAME>"
+```
+
+#### A bit about network history
+The network history feature allows you to get data into your data node (which acts as a rich API node) faster by fetching data from the other nodes instead of replaying it.
+
+There is an entire section called `NetworkHistory` in the data node's `config.toml`. To use it, you must provide at least one other data node that exposes the IPFS node in the `NetworkHistory.Store.BootstrapPeers` parameter. If you do not provide that, you won't be able to use the IPFS-based network history.
+
+Format of `BootstrapPeers`:
+
+```
+"/dns/<DATA-NODE-HOST>/tcp/<DATA-NODE-SWARM-PORT>/ipfs/<DATA-NODE-PEER-ID>"
+```
+
+Example `BootstrapPeers` value:
+
+**Do not use the below bootstrap peers as it is an example from a test network.**
```

Comment:
Would it be worth mentioning you can hit a peers network history rest endpoint to source peer info? e.g

https://api.n00.testnet.vega.xyz/api/v2/networkhistory

yields:

`{"ipfsAddress":"/ip4/164.92.198.87/tcp/4001/p2p/12D3KooWNiWcT93S3P3eiHqGq4a6feaD2cUfbWw9AxgdVt8RzTHJ","swarmKey":"/key/swarm/psk/1.0.0/\n/base16/\n766567612d6661697267726f756e642d32303233303432363135313900000000","swarmKeySeed":"vega-fairground-202304261519","connectedPeers":["/ip4/159.203.12.201/tcp/4001/p2p/12D3KooWACJuzchZQH8Tz1zNmkGCatgcS2DUoiQnMFaALVMo7DpC","/ip4/139.59.0.80/tcp/4001/p2p/12D3KooWMSaQevxg1JcaFxWTpxMjKw1J13bLVLmoxbeSJ5gpXjRh","/ip4/104.248.42.237/tcp/4001/p2p/12D3KooWDy26YcZVTPbYCmZu3Jukp7Ygjf33ksXTV6GENNowiqBa","/ip4/104.248.42.237/tcp/51338/p2p/12D3KooWDy26YcZVTPbYCmZu3Jukp7Ygjf33ksXTV6GENNowiqBa","/ip4/139.59.0.80/tcp/58318/p2p/12D3KooWMSaQevxg1JcaFxWTpxMjKw1J13bLVLmoxbeSJ5gpXjRh"]}`

the value of ipfsAddress field in the above output can be used as a bootstrap peer

Reply:
This looks good, but if someone is in the process of upgrading, will this URL work for mainnet? Or do they all need to upgrade before this is available and thus they can only use it after this version?

Reply:
@ettec hopefully something we can remember to improve in the docs for a future upgrade